recipes/cpp-httplib/all/conanfile.py

```diff
@@ -20,17 +21,27 @@ class CpphttplibConan(ConanFile):
         "with_openssl": [True, False],
         "with_zlib": [True, False],
         "with_brotli": [True, False],
+        "use_macos_keychain_certs": [True, False],
     }
     default_options = {
         "with_openssl": False,
         "with_zlib": False,
         "with_brotli": False,
+        "use_macos_keychain_certs": True,
     }
     no_copy_source = True
 
+    def config_options(self):
+        if self.settings.os != "Macos":
+            del self.options.use_macos_keychain_certs
+
     def requirements(self):
         if self.options.with_openssl:
-            self.requires("openssl/[>=1.1 <4]")
+            if Version(self.version) < "0.15":
+                self.requires("openssl/[>=1.1 <4]")
+            else:
+                # New version of httplib.h requires OpenSSL 3
+                self.requires("openssl/[>=3 <4]")
```

Comment:
Activating openssl for newer versions triggered 
`httplib.h:273:2: error: Sorry, OpenSSL versions prior to 3.0.0 are not supported` for all versions after 0.15